fw/http.c

```diff
@@ -2591,6 +2591,20 @@ tfw_http_conn_msg_alloc(TfwConn *conn, TfwStream *stream)
 	if (unlikely(!hm))
 		return NULL;
 
+	if (TFW_FSM_TYPE(conn->proto.type) != TFW_FSM_H2 && type & Conn_Clnt) {
+		TfwHttpReq *req_prev;
+		TfwCliConn *cli_conn = (TfwCliConn *)conn;
+		struct list_head *seq_queue = &cli_conn->seq_queue;
+
+		req_prev = list_empty(seq_queue) ? NULL :
+			list_last_entry(seq_queue, TfwHttpReq, msg.seq_list);
```

Comment:
As alternative this can be done via introducing field `flags` in `TfwCliConn` struct;

Reply:
I'm OK with the check, but I think `tfw_http_conn_msg_alloc()` isn't a good place for the websocket logic - we shouldn't call the function if we already have Upgrade request. The only path for the function is `tfw_http_req_process()` -> `tfw_h1_req_process()` -> `tfw_http_msg_create_sibling()` (the 2nd call of the function is from `tfw_http_resp_process()`, so it's our of our scope)  - in this path we already have the patch on line 5808 (we create sibling from a split skb). The other call `tfw_http_conn_msg_alloc()` is from `tfw_http_msg_process_generic()` and probably this is the place for this code (it already has websocket logic BTW).

Reply:
No, the path `tfw_http_req_process() -> tfw_h1_req_process() -> tfw_http_msg_create_sibling()` it's not single path, it's only for pipelined requests that came in single SKB. And yes this path must contain check, but by mistake I moved the check to response. I will add check to this path, however we can't remove check from `tfw_http_conn_msg_alloc()`, because requests can come to tempesta in separated SKBs.